### PR TITLE
fix(lambda): remove template path from sam deploy command

### PR DESCRIPTION
**Pull request by vicheey**
## Problem
Explicitly specifying template path in sam deploy command to point to the build artifact folder makes SAM CLI create/update `samconfig.toml` in the build artifact folder rather than updating the `samconfig.toml` file in project root folder. 

## Solution
Remove the `--template-path` from deploy flag. SAM will determine the template in the project folder and write to correct samconfig file

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by roger-zhangg — APPROVED** [review]
LGTM

File: packages/toolkit/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 3.33.0 2024-11-06
+- **Bug Fix** SAM: Save SAM deploy parameters to correct samconfig.toml in project folder
+
```
Comment (hayemaxi) [inline packages/toolkit/CHANGELOG.md:3]: Do not manually add the changelog. Use `npm run newChange`

Comment (vicheey) [reply packages/toolkit/CHANGELOG.md]: I reverted manual update and replaced with change json object for next release.

**Review by hayemaxi — APPROVED** [review]
lgtm pending integration tests

**Comment by vicheey** [comment]
Looking into Intg test failures

**Review by hayemaxi — CHANGES_REQUESTED** [review]
integ tests need fixing

**Comment by vicheey** [comment]
The integration tests failures do not seem to be related to the new change introduced in this PR but more like false negative resulted from issue with mocking.

**Review by seshubaws — APPROVED** [review]

**Pull request merged**